packages/commons/test/FileAxeScriptProviderTest.cs

```diff
@@ -0,0 +1,69 @@
+using NUnit.Framework;
+using Deque.AxeCore.Commons;
+using Moq;
+using FluentAssertions;
+using System;
+using System.IO;
+using System.Runtime.InteropServices;
+
+namespace Deque.AxeCore.Commons.Test
+{
+    [TestFixture]
+    [NonParallelizable]
+    public class FileAxeScriptProviderTest
+    {
+        private static readonly string testContext = TestContext.CurrentContext.TestDirectory.ToString();
+        private readonly string basePath = testContext.Split(new string[] { "test" }, StringSplitOptions.None)[0];
+        
+        [Test]
+        public void ConstructorPassedValidFile()
+        {
+            var absolutePath = Path.Combine(basePath, "src", "Resources", "sampleFile.txt");
+            
+            var scriptProvider = new FileAxeScriptProvider(absolutePath);
+            scriptProvider.Should().NotBeNull();
+            var readResult = scriptProvider.GetScript();
+            readResult.Should().Be("sample output");
+        }
+
+        [Test]
+        public void ConstructorPassedNullOrEmptyString()
+        {
+            Assert.Throws<ArgumentNullException>(() =>
+            {
+                var scriptProvider = new FileAxeScriptProvider(null);                
+                scriptProvider.Should().NotBeNull();
+            });
+            
+            Assert.Throws<ArgumentNullException>(() =>
+            {
+                var scriptProvider = new FileAxeScriptProvider("");                
+                scriptProvider.Should().NotBeNull();
+            });
+        }
+
+        [Test]
+        public void ConstructorPassedNonexistentFile()
+        {
+            Assert.Throws<ArgumentException>(() =>
+            {
+                var scriptProvider = new FileAxeScriptProvider("sample.html");
+                scriptProvider.Should().NotBeNull();
+            });
+        }
+
+        [Test]
+        public void GetScriptForNonexistentFile()
+        {
+            var testFilePath = Path.Combine(basePath, "src", "Resources", "foo.txt");
```

Comment:
In general, it's best to avoid using a source path for a temporary test file like this - if the test ever happens to fail between the create and delete steps, you don't want to end up with extra files being tracked by git. Temporary test files should always go to some sort of `.gitignore`'d working directory. In our case, since we can have multiple instances of the tests running for each target framework, we also want to avoid reusing the same directory between different target frameworks so the tests won't step on each other if the wrong ones happen to run in parallel.

I'd recommend doing something like `Path.Combine(typeof(FileAxeScriptProviderTest).Assembly.Location, "..", "foo.txt")` - the test assembly location is already a `.gitignore`'d output directory which is different for each framework.